src/dev/Web/EastSeat.ResourceIdea.Web/Components/Pages/Backend/Users.razor

```diff
@@ -0,0 +1,346 @@
+                                            </button>
+                                            @if (!user.IsBackendUser)
+                                            {
+                                                <button class="btn btn-outline-warning" @onclick="() => ImpersonateUser(user.Id)">
+                                                    <i class="fas fa-user-secret"></i>
+                                                </button>
+                                            }
+                                            <button class="btn btn-outline-danger" @onclick="() => ToggleUserStatus(user.Id)">
+                                                <i class="fas @(user.IsActive ? "fa-user-slash" : "fa-user-check")"></i>
+                                            </button>
+                                        </div>
+                                    </td>
+                                </tr>
+                            }
+                        </tbody>
+                    </table>
+                </div>
+
+                @if (!filteredUsers.Any())
+                {
+                    <div class="text-center py-5">
+                        <i class="fas fa-users fa-3x text-muted mb-3"></i>
+                        <h5 class="text-muted">No users found</h5>
+                        <p class="text-muted">Try adjusting your search criteria or create a new user.</p>
+                    </div>
+                }
+            </div>
+        </div>
+
+        <!-- Pagination -->
+        <nav aria-label="User pagination" class="mt-4">
+            <ul class="pagination justify-content-center">
+                <li class="page-item @(currentPage <= 1 ? "disabled" : "")">
+                    <button class="page-link" @onclick="() => GoToPage(currentPage - 1)">Previous</button>
+                </li>
+                
+                @for (int i = Math.Max(1, currentPage - 2); i <= Math.Min(totalPages, currentPage + 2); i++)
+                {
+                    <li class="page-item @(i == currentPage ? "active" : "")">
+                        <button class="page-link" @onclick="() => GoToPage(i)">@i</button>
+                    </li>
+                }
+                
+                <li class="page-item @(currentPage >= totalPages ? "disabled" : "")">
+                    <button class="page-link" @onclick="() => GoToPage(currentPage + 1)">Next</button>
+                </li>
+            </ul>
+        </nav>
+    }
+</div>
+
+@code {
+    private List<UserListItem> allUsers = new();
+    private List<UserListItem> filteredUsers = new();
+    private string searchTerm = string.Empty;
+    private int currentPage = 1;
+    private int pageSize = 20;
+    private int totalPages = 1;
+
+    protected override async Task OnInitializedAsync()
+    {
+        await ExecuteAsync(async () =>
+        {
+            await LoadUsers();
+        }, "Loading users");
+    }
+
+    private async Task LoadUsers()
```

Comment:
Don't simulate loading of users. Reference `TenantEmployeesQuery` and `TenantEmployeesQueryHandler` to implement a query and handler that will get the users that have been setup in the system.